[이다빈] Mission 3 : 회원가입 기능 구현(2차 수정)

thisisempty opened the pull request:
westagram/settings.py : Insatalled_apps에 'users'추가
westagram/urls.py : users app path추가

[CREATE]
users/views.py : SignUpView - post 생성
users/urls.py : SignUpView url path 생성

>wecode PR message template 및 체크 리스트 입니다. 
아래 사항들을 전부 작성/체크 하시고 PR 해주세요!

## 수정 사항 간략한 한줄 요약
(여기에 수정 사항에 대한 간략한 한줄 요약/제목 작성해 주세요.)
- 회원가입 기능 구현
		 
## 수정 사항들 자세한 내용
(여기에 수정 사항에 대한 자세한 내용을 작성해 주세요.)
- phone column max_length 20에서 100으로 변경
- 코딩 컨벤션에 맞추어 수정

## 기타 질문 및 특이 사항
(궁금한 사항들, 하면서 느낀 점들 공유해주실 것이 있으면 자유롭게 작성해 주세요.)
- 장고의  vaildate_email쓰지 않고 regex를 쓰는 이유가 궁금합니다.

## 체크 리스트 (아래 사항들이 전부 체크되어야만 merge가 됩니다!)
조건을 만족한다면 각 빈칸에 x 표시 해주세요
- [x] 필요한 test들을 완료하였고 기능이 제대로 실행되는지 확인 하였습니다.
- [x] Wecode의 코드 스타일 가이드에 맞추어 코드를 작성 하였습니다.
- [x] 제가 의도한 파일들과 수정 사항들만 커밋이 된 것을 확인 하였습니다.
- [x] 본 수정 사항들을 팀원들과 사전에 상의하였고 팀원들 모두 해당 PR에 대하여 알고 있습니다.

thisisempty committed:
[ADD]
westagram/settings.py : Insatalled_apps에 'users'추가
westagram/urls.py : users app path추가

[CREATE]
users/views.py : SignUpView - post 생성
users/urls.py : SignUpView url path 생성

thisisempty committed:
modify User.password max_length 45 -> 200

thisisempty committed:
[MODIFY]
users/models.py : Change the phone column from an IntegerField to a CharField
users/views.py : if not email or not password부분이 except의 기능과 동일한 것 같아 삭제.

kyeongraekim commented on students/dabeen/westagram/users/views.py:
```diff
@@ -1,3 +1,40 @@
+
+        try:
+            data = json.loads(request.body)
+
```
중간 줄바꿈 공백이 불필요해보이네요~ 해당 공백 제거해주세요.

thisisempty replied:
수정 후 커밋했습니다.

kyeongraekim commented on students/dabeen/westagram/users/views.py:
```diff
@@ -1,3 +1,40 @@
+            )
+
+            return JsonResponse({'message': 'SUCCESS'}, status=201)
+        except KeyError:
```
except 문 시작시 한 줄 띄어서 try 문과 차이를 두는게 좋을 것 같습니다.

thisisempty replied:
수정 후 커밋했습니다.

kyeongraekim commented on students/dabeen/westagram/users/models.py:
```diff
@@ -1,12 +1,13 @@
     email = models.CharField(max_length=45, unique=True)
     password = models.CharField(max_length=200)
-    phone = models.IntegerField()
+    phone = models.CharField(max_length=20)
```
전세계 사용자를 대상으로 한 인별그램 기획상 나라별 국제번호 및 번호 패턴이 상이하다는 점, 기타 특수기호가 들어갈 가능성을 염두하여 max_length에 조금 더 여유를 두셔도 좋을 것 같습니다.

thisisempty replied:
수정 후 커밋했습니다.

kyeongraekim commented on students/dabeen/westagram/users/views.py:
```diff
@@ -1,3 +1,40 @@
+import json
+import re
+from django.http import JsonResponse
+from django.views import View
```
여러 모듈, 라이브러리를 import하는 순서도 그 특성에 맞게 개발자간 일종의 약속처럼 정해져있습니다.
위 views.py 에서 import 하는 내용들의 순서와 줄바꿈 공백을 위코드 컨벤션에 맞게 수정해주세요~

thisisempty replied:
수정 후 커밋했습니다.

thisisempty committed:
[MODIFY]

users/models.py : phone column's max_length 20 -> 100
users/views.py : modifying according to the coding convention

lordmyshepherd commented:
LGTM 👍

lordmyshepherd commented on students/dabeen/westagram/users/views.py:
```diff
@@ -1,3 +1,40 @@
+            if User.objects.filter(email=email).exists():
+                return JsonResponse({'message': 'EMAIL_ALREADY_EXISTS'}, status=400)
+
+            regex_email = '^[a-zA-Z0-9+-_.]+@[a-zA-Z0-9-]+\.[a-zA-Z0-9-.]+$'
+            regex_password = '^(?=.*[A-Z])(?=.*[a-z])(?=.*[0-9]).{8,}'
```
상수는 대문자로 만들어주세요.

```suggestion
            REGEX_EMAIL = '^[a-zA-Z0-9+-_.]+@[a-zA-Z0-9-]+\.[a-zA-Z0-9-.]+$'
            REGEX_PASSWORD = '^(?=.*[A-Z])(?=.*[a-z])(?=.*[0-9]).{8,}'
```

thisisempty replied:
수정 후 커밋 했습니다.

thisisempty committed:
modify constant from lowercase to uppercase
